apps/theme/layouts/forms/DefaultFormLayout.tsx

```diff
@@ -73,17 +104,44 @@ const DefaultFormLayout: FormLayoutComponent = ({
                             ))}
                         </Row>
                     ))}
-
                     {termsOfServiceEnabled && <TermsOfServiceSection component={TermsOfService} />}
                     {reCaptchaEnabled && <ReCaptchaSection component={ReCaptcha} />}
-
-                    <SubmitButton
-                        onClick={submit}
-                        loading={loading}
-                        fullWidth={formData.settings.fullWidthSubmitButton}
-                    >
-                        {formData.settings.submitButtonLabel || "Submit"}
-                    </SubmitButton>
+                    {/*
+                        If the form has more than one step then the form will be recognized as a Multi Step Form,
+                        so it means that we need to render form step handlers to switch between steps.
+                    */}
+                    {isMultiStepForm && (
+                        <ButtonsWrapper>
+                            <DefaultButton onClick={goToPreviousStep} disabled={currentStep === 0}>
+                                Previous Step
+                            </DefaultButton>
+                            {currentStep === formData.steps.length - 1 ? (
+                                <SubmitButton onClick={submit} loading={loading} fullWidth={false}>
+                                    {formData.settings.submitButtonLabel || "Submit"}
+                                </SubmitButton>
+                            ) : (
+                                <SubmitButton
+                                    onClick={() => {
+                                        validateCurrentStepFields(form);
```

Comment:
@adrians5j 
Actually I do that extra validation because without it the method `goToNextStep` would always work and the form would be submited (even though validation failed). Maybe we can pass `goToNextStep` method inside of the `Form` component and call it there?

Reply:
Can't we just call `submit` here, and, in the `onSubmit` callback (which will be called after the `<Form>` component has validated the rendered fields) do the next step? Since it's defined internally, the `onSubmit` callback can internally do all the logic of "go next step" or "finish form submission - show success". 

I hope I'm making sense. :)

Reply:
Yeah, you're right!
I just did not realize that we could do that in the `onSubmit`.